wrangler-service/src/main/java/co/cask/wrangler/service/kafka/KafkaConfiguration.java

```diff
@@ -95,10 +97,32 @@ public KafkaConfiguration(Connection conn, Map<String, String> runtimeArgs) {
     props.put(ConsumerConfig.VALUE_DESERIALIZER_CLASS_CONFIG, valueDeserializer);
     props.put(ConsumerConfig.EXCLUDE_INTERNAL_TOPICS_CONFIG, kafkaProducerProperties.getOrDefault(ConsumerConfig.EXCLUDE_INTERNAL_TOPICS_CONFIG, "true"));
     props.put(ConsumerConfig.REQUEST_TIMEOUT_MS_CONFIG, kafkaProducerProperties.getOrDefault(ConsumerConfig.REQUEST_TIMEOUT_MS_CONFIG, "15000"));
+    if (!Strings.isNullOrEmpty(properties.get(SchemaRegistryClient.Configuration.SCHEMA_REGISTRY_URL.name())) && "avro".equalsIgnoreCase(properties.get(PropertyIds.FORMAT))) {
+    	keyDeserializer = ByteArrayDeserializer.class.getCanonicalName();
+    	valueDeserializer = KafkaAvroDeserializer.class.getCanonicalName();
+    	props.setProperty(PropertyIds.SCHEMA_NAME, properties.getOrDefault(PropertyIds.SCHEMA_NAME, properties.get(PropertyIds.TOPIC)));
+    	props.put(SchemaRegistryClient.Configuration.SCHEMA_REGISTRY_URL.name(), properties.get(SchemaRegistryClient.Configuration.SCHEMA_REGISTRY_URL.name()));
+    	LOG.info("Schema Registry is provided, picking up provided URL as: {}", properties.get(SchemaRegistryClient.Configuration.SCHEMA_REGISTRY_URL.name()));
+    } else if ("binary".equalsIgnoreCase(properties.get(PropertyIds.FORMAT)) || "avro".equalsIgnoreCase(properties.get(PropertyIds.FORMAT))) {
+    	keyDeserializer = ByteArrayDeserializer.class.getCanonicalName();
+    	valueDeserializer = ByteArrayDeserializer.class.getCanonicalName();
+    }
     
-    String keytabLocation = properties.getOrDefault(PropertyIds.KEYTAB_LOCATION, runtimeArgs.get(PropertyIds.NAMESPACE_KETAB_PATH));
-    String principal = properties.getOrDefault(PropertyIds.PRINCIPAL, runtimeArgs.get(PropertyIds.NAMESPACE_PRINCIPAL_NAME));
-
+    keyDeserializer = kafkaProducerProperties.getOrDefault(ConsumerConfig.KEY_DESERIALIZER_CLASS_CONFIG, keyDeserializer);
+    valueDeserializer = kafkaProducerProperties.getOrDefault(ConsumerConfig.VALUE_DESERIALIZER_CLASS_CONFIG, valueDeserializer);
+    
+    props.put(ConsumerConfig.KEY_DESERIALIZER_CLASS_CONFIG, keyDeserializer);
+    props.put(ConsumerConfig.VALUE_DESERIALIZER_CLASS_CONFIG, valueDeserializer);
+    
+    keytabLocation = properties.get(PropertyIds.KEYTAB_LOCATION);
+    if (Strings.isNullOrEmpty(keytabLocation) && runtimeArgs.containsKey(PropertyIds.NAMESPACE_KETAB_PATH)) {
+    	String[] arr = runtimeArgs.get(PropertyIds.NAMESPACE_KETAB_PATH).split("/");
+    	keytabLocation = "./" + arr[arr.length - 1];
+    }
+    principal = properties.get(PropertyIds.PRINCIPAL);
+    if (Strings.isNullOrEmpty(principal)) {
+    	principal = runtimeArgs.get(PropertyIds.NAMESPACE_PRINCIPAL_NAME);
+    }
```

Comment:
There could be an issue where user specifies one argument say 'keytab' and other argument 'principal' is empty. In this case, you will use Namespace principal but keytab specified by user. We should throw error in such cases. Either both user specified arguments should be picked or both Namespace preference arguments